selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/TheiaIde.java

```diff
@@ -0,0 +1,120 @@
+import com.google.inject.Singleton;
+import org.eclipse.che.selenium.core.SeleniumWebDriver;
+import org.eclipse.che.selenium.core.webdriver.SeleniumWebDriverHelper;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.PageFactory;
+
+@Singleton
+public class TheiaIde {
+
+  private final SeleniumWebDriverHelper seleniumWebDriverHelper;
+
+  @Inject
+  TheiaIde(SeleniumWebDriver seleniumWebDriver, SeleniumWebDriverHelper seleniumWebDriverHelper) {
+    this.seleniumWebDriverHelper = seleniumWebDriverHelper;
+    PageFactory.initElements(seleniumWebDriver, this);
+  }
+
+  public interface Locators {
+    String THEIA_IDE_ID = "theia-app-shell";
+    String THEIA_IDE_TOP_PANEL_ID = "theia-top-panel";
+    String LOADER_XPATH = "//div[@class='theia-preload theia-hidden']";
+    String MAIN_MENU_ITEM_XPATH_PATTERN = "//div[@id='theia:menubar']//li/div[text()='%s']";
+    String SUBMENU_ITEM_XPATH_PATTERN = "//li[@class='p-Menu-item']/div[text()='%s']";
+    String ABOUT_DIALOG_XPATH = "//div[@class='dialogBlock']";
+    String ABOUT_DIALOG_TITLE_XPATH = ABOUT_DIALOG_XPATH + "//div[@class='dialogTitle']";
+    String ABOUT_DIALOG_CONTENT_XPATH = ABOUT_DIALOG_XPATH + "//div[@class='dialogContent']";
+    String ABOUT_DIALOG_OK_BUTTON_XPATH = ABOUT_DIALOG_XPATH + "//button";
+  }
+
+  @FindBy(id = Locators.THEIA_IDE_ID)
+  WebElement theiaIde;
+
+  @FindBy(id = Locators.THEIA_IDE_TOP_PANEL_ID)
+  WebElement theiaIdeTopPanel;
+
+  @FindBy(xpath = Locators.LOADER_XPATH)
+  WebElement loader;
+
+  @FindBy(xpath = Locators.ABOUT_DIALOG_XPATH)
+  WebElement aboutDialog;
+
+  @FindBy(xpath = Locators.ABOUT_DIALOG_TITLE_XPATH)
+  WebElement aboutDialogTitle;
+
+  @FindBy(xpath = Locators.ABOUT_DIALOG_CONTENT_XPATH)
+  WebElement aboutDialogContent;
+
+  @FindBy(xpath = Locators.ABOUT_DIALOG_OK_BUTTON_XPATH)
+  WebElement aboutDialogOkButton;
+
+  public void waitTheiaIde() {
+    seleniumWebDriverHelper.waitVisibility(theiaIde, PREPARING_WS_TIMEOUT_SEC);
+  }
+
+  public void waitTheiaIdeTopPanel() {
+    seleniumWebDriverHelper.waitVisibility(theiaIdeTopPanel);
+  }
+
+  public void waitLoaderInvisibility() {
+    seleniumWebDriverHelper.waitInvisibility(loader);
+  }
+
+  public void clickOnMenuItemInMainMenu(String itemName) {
+    seleniumWebDriverHelper.waitAndClick(
+        By.xpath(format(Locators.MAIN_MENU_ITEM_XPATH_PATTERN, itemName)));
+  }
+
+  public void clickOnSubmenuItem(String itemName) {
+    seleniumWebDriverHelper.waitAndClick(
+        By.xpath(format(Locators.SUBMENU_ITEM_XPATH_PATTERN, itemName)));
+  }
+
+  /**
+   * Run command from sub menu.
+   *
+   * @param topMenuCommand
+   * @param commandName
+   */
+  public void runCommand(String topMenuCommand, String commandName) {
+    clickOnMenuItemInMainMenu(topMenuCommand);
+    clickOnSubmenuItem(commandName);
+  }
+
+  /** wait 'About' dialog is open */
```

Comment:
`/** wait 'About' dialog is open */` looks redundant so as it is just repeat method name.